pkg/util/util.go

```diff
@@ -0,0 +1,26 @@
+package util
+
+import (
+	"strings"
+
+	"github.com/openshift/csi-driver-shared-resource/pkg/config"
+)
+
+// IsAuthorizedSharedResource verify the shared resource already exists under OCP pre-populated shared resource
+func IsAuthorizedSharedResource(resource, kind, resourceRefKeyNm, resourceRefKeyNs string) bool {
+	var (
+		value  string
+		exists bool
+	)
+	if strings.HasPrefix(resource, "openshift-") {
+		if strings.EqualFold(kind, "SharedSecret") {
+			value, exists = config.LoadedSharedSecretConfig.AuthorizedSharedResources[resource]
+		} else {
+			value, exists = config.LoadedSharedConfig.AuthorizedSharedResources[resource]
+		}
```

Comment:
Let's use a switch/case statement here to facilitate adding more types in the future if needed